eth/backend.go

```diff
@@ -486,6 +488,47 @@ func New(ctx context.Context, stack *node.Node, config *ethconfig.Config, logger
 		}()
 	}
 
+	// setup periodic logging and prometheus updates
+	go func() {
```

Comment:
can wrap this into a function `LogVirtualMemStats` and move it in the new erigon-lib/common/mem package you have added

here in init code you can just do `go mem.LogVirtualMemStats()`